libpod/boltdb_state_internal.go

```diff
@@ -847,3 +855,178 @@ func (s *BoltState) removeContainer(ctr *Container, pod *Pod, tx *bolt.Tx) error
 
 	return nil
 }
+
+// Evicts a container from the DB
+// If the config specifies a pod, the container is treated as belonging to a pod,
+// and will be removed from the pod as well
+func (s *BoltState) evictContainer(ctrID []byte, config *ContainerConfig, depCtrs []string, tx *bolt.Tx) error {
```

Comment:
Hmmm. I think we can reuse `RemoveContainer()` here - it doesn't actually need anything in the container state.

We can make a Container struct with a ContainerConfig we get out of GetContainerConfig, set state to an empty struct to avoid potential dereference segfaults, and pass it into RemoveContainer

Reply:
That makes sense, I have done some more refactoring and removed a good part of the duplicated code.